keps/sig-api-machinery/2885-server-side-unknown-field-validation/README.md

```diff
@@ -677,51 +717,46 @@ only new requests to the api-server will trigger strict validation.
 
 ###### Are there any tests for feature enablement/disablement?
 
-Testing for both presence and absence of query param.
-
-### Rollout, Upgrade and Rollback Planning
+Testing for both presence and absence of query param when the feature is enabled. We also test
+that modified codepaths that could be triggered when the feature is enabled are
+inert when the feature is disabled.
 
-This section must be completed when targeting beta to a release.
+For example, unknown field path tracking in both the unstructured converter and
+structural pruning algorithm affect performance when server-side field
+validation is performed, but are tested to ensure that they are never invoked
+when server side field validation is disabled.
 
-N/A
+### Rollout, Upgrade and Rollback Planning
 
-<!--
 ###### How can a rollout or rollback fail? Can it impact already running workloads?
 
-Try to be as paranoid as possible - e.g., what if some components will restart
-mid-rollout?
+API servers rolled out with field validation enabled could start erroneously
+failing requests if there are problems with how field validation is implemented.
 
-Be sure to consider highly-available clusters, where, for example,
-feature flags will be enabled on some API servers and not others during the
-rollout. Similarly, consider large clusters and how enablement/disablement
-will rollout across nodes.
+Also, performance issues could slow down API server performance or memory usage
+if extraneous validation is being invoked.
 
 ###### What specific metrics should inform a rollback?
 
-What signals should users be paying attention to when the feature is young
-that might indicate a serious problem?
+A drastic deterioration in API server performance could indicate an issue with
+the feature and a need to rollback (i.e. `apiserver_request_duration_seconds`).
 
 ###### Were upgrade and rollback tested? Was the upgrade->downgrade->upgrade path tested?
 
-Describe manual testing that was done and the outcomes.
-Longer term, we may want to require automated upgrade/rollback tests, but we
-are missing a bunch of machinery and tooling and can't do that now.
+No
 
 ###### Is the rollout accompanied by any deprecations and/or removals of features, APIs, fields of API types, flags, etc.?
 
-Even if applying deprecation policies, they may still surprise some users.
--->
-### Monitoring Requirements
+For beta we will be warning kubectl users that `--validate={true,false}` is
```

Comment:
why do we need to deprecate --validate=true/false when we can make --validate=true use server-side validation on servers that support it automatically?

Reply:
That's fair. My thinking on the deprecation is that long term validation will be expressed in terms of "Strict", "Warn", and "Ignore". I guess there's no real reason to deprecate "true/false" other than that it might confuse people who expect the behavior of "--validate=true" to be the old client-side behavior. Removed the note about deprecating it.